### PR TITLE
disableable assertions

### PR DESCRIPTION
**Pull request by ofrobots**
state.js needs some assertions because it interfaces with parts of V8
debug API that are volatile across versions. However, we don't want to
be running these assertions in real user code. debug-assert module
allows us to run with assertions enabled in the CI.

### CODE REVIEW

File: bin/run-test.sh [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 # Usage: -c to report coverage
 
+# Enable assertions
+export DEBUG_AGENT_ASSERTIONS=1
```
Comment (matthewloring) [inline bin/run-test.sh]: We've used the structure `CLOUD_DEBUG_xxxx` for our other env vars. Might be good to keep things consistent.

File: test/test-debug-assert.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+describe('debug-assert', () => {
+  const debugAssert = require('../src/agent/debug-assert.js');
+
+  it('should not fire assertions when disabled', () => {
+    const assert = debugAssert(false);
+    assert.equal(1, 2);
+  });
+
+  it('should fire assertions when enabled', (done) => {
```
Comment (matthewloring) [inline test/test-debug-assert.js]: Could we add a test to ensure full assert api coverage?

File: bin/run-test.sh (hunk 1 above)
Comment (ofrobots) [inline bin/run-test.sh]: Done.

File: test/test-debug-assert.js (hunk 2 above)
Comment (ofrobots) [inline test/test-debug-assert.js]: Done.

Added another type of test to ensure that CI is running with assertions enabled.

**Comment by matthewloring** [comment]
I think we also need the environment variable set on mocha.

**Comment by ofrobots** [comment]
> I think we also need the environment variable set on mocha.

Yes. Users running mocha manually (rather than run-test.sh) will need to make sure that the env. var. is defined. The purpose of the test is the make sure that our CI always runs with the env. var. is defined.

I am not sure if you are requesting a change or making an observation. Can you approve or elaborate on what change you would like to see?

**Comment by matthewloring** [comment]
Complete mistype. Meant to say appveyor. Should the environment variable be set on appveyor?

**Comment by ofrobots** [comment]
Ah, so appveyor doesn't use run-test.sh? Ofcourse it doesn't. Dx

**Comment by ofrobots** [comment]
PTAL.

**Pull request merged**